app/src/main/java/space/taran/arknavigator/mvp/presenter/adapter/tagsselector/TagsSelectorPresenter.kt

```diff
@@ -187,6 +203,22 @@ class TagsSelectorPresenter(
         viewState.drawTags()
     }
 
+    private fun getVirtualTags(
+        tagsByResources: Map<ResourceId, Tags>
+    ): Map<ResourceId, Tags> {
+        var modifiedTagsByResources = tagsByResources
+        if (showKindTags) {
+            modifiedTagsByResources = modifiedTagsByResources.map { (id, tags) ->
+                val listOfTags = tags.toMutableList()
+                if (kindByExt(index!!.getMeta(id).extension) != null) {
+                    listOfTags.add(KIND + kindByExt(index!!.getMeta(id).extension))
```

Comment:
Just `tags.filter` instead of using mutable collections.

Reply:
updated in new PR